mobile/src/main/java/org/openhab/habdroid/core/OpenHABVoiceService.java

```diff
@@ -9,162 +9,64 @@
-     * Therefore all intents are buffered and later this buffer will be processed.
-     *
-     * @param intent The {@link Intent} to be buffered.
-     */
-    private void bufferIntent(Intent intent) {
-        mBufferedIntents.add(intent);
-    }
-
-    @Override
-    public void onOpenHABTracked(String baseUrl) {
-        Log.d(TAG, "onOpenHABTracked(): " + baseUrl);
-        mOpenHABBaseUrl = baseUrl;
-        while (!mBufferedIntents.isEmpty()) {
-            processVoiceIntent(mBufferedIntents.poll());
-        }
-        Log.d(TAG, "Stopping service for start ID " + getLastStartId());
-        stopSelf(getLastStartId());
-    }
-
-    @Override
-    public void showMessageToUser(String message, int messageType, int logLevel) {
-        showMessageToUser(message, messageType, logLevel);
-    }
-
-    /**
-     * @param message message to show
-     * @param messageType must be Constants.MESSAGES.DIALOG or Constants.MESSAGES.TOAST
-     * @param logLevel not implemented
-     */
-    public void showMessageToUser(String message, int messageType, int logLevel, int actionMessage, @Nullable View.OnClickListener actionListener) {
-        if(message == null) {
-            return;
-        }
-        switch (messageType) {
-            case Constants.MESSAGES.DIALOG:
-                AlertDialog.Builder builder = new AlertDialog.Builder(OpenHABVoiceService.this);
-                builder.setMessage(message)
-                        .setPositiveButton("OK", new DialogInterface.OnClickListener() {
-                            public void onClick(DialogInterface dialog, int id) {
-                            }
-                        });
-                AlertDialog alert = builder.create();
-                alert.show();
-                break;
-            default:
-                Toast.makeText(this, message, Toast.LENGTH_LONG).show();
-                break;
-        }
-    }
-
-    @Override
-    public void onError(String error) {
-        showMessageToUser(error, Constants.MESSAGES.DIALOG, Constants.MESSAGES.LOGLEVEL.ALWAYS);
-        Log.d(TAG, "onError(): " + error);
-        stopSelf();
-    }
-
-
-    private void processVoiceIntent(Intent data) {
-        Log.d(TAG, "processVoiceIntent()");
-
-        String voiceCommand = extractVoiceCommand(data);
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        String voiceCommand = extractVoiceCommand(intent);
+        boolean hasSentCommand = false;
         if (!voiceCommand.isEmpty()) {
-            if (mOpenHABBaseUrl != null) {
-                sendItemCommand("VoiceCommand", voiceCommand);
-            } else {
-                Log.w(TAG, "Couldn't determine OpenHAB URL");
+            try {
+                Connection conn = ConnectionFactory.getUsableConnection();
+                if (conn != null) {
+                    sendItemCommand("VoiceCommand", voiceCommand, conn, startId);
+                    hasSentCommand = true;
+                }
```

Comment:
I think there's an else path missing here (which registers a connection change listener and memorizes the intent so it can be reevaluated when the listener is fired); as otherwise VoiceService won't work if the app was not running before its invocation.

Reply:
If I understood you correctly, this will be included in the next commit :)